### ISSUE TITLE
Libp2p ping error

### ISSUE THREAD

**Issue by kingalg** [TryQuiet/quiet#2108]
Desktop: Version: 2.0.3-alpha.6

Libp2p ping error

On the Owner side:

```
backend:Libp2pService QmNcQbPejv6MDBt9VVt7A9uSU1a2yiYnhHsnVo2hhfqpFL is connected to 1 peers +0ms
  backend:Libp2pService QmNcQbPejv6MDBt9VVt7A9uSU1a2yiYnhHsnVo2hhfqpFL has 2 open connections +0ms
node:internal/event_target:1006
  process.nextTick(() => { throw err; });
                           ^

Error: stream reset
    at Object.reset (/Applications/Quiet.app/Contents/Resources/app.asar/node_modules/backend-bundle/bundle.cjs:431809:34)
    at MplexStreamMuxer._handleIncoming (/Applications/Quiet.app/Contents/Resources/app.asar/node_modules/backend-bundle/bundle.cjs:432184:24)
    at MplexStreamMuxer.sink (/Applications/Quiet.app/Contents/Resources/app.asar/node_modules/backend-bundle/bundle.cjs:432087:36)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Promise.all (index 0) {
  code: 'ERR_STREAM_RESET'
```

<img width="244" alt="Screenshot 2023-11-22 at 14 03 51" src="https://github.com/TryQuiet/quiet/assets/68909168/ae574cd7-c3e1-4c63-b5fc-1f9eee2cca6c">

On user side (this user joined the owner and error appeared when another user tried to join from his invite code):
```
2023-11-22T12:55:33.625Z backend:Libp2pService QmYFmZvEikCwJpCjZ6FAiXnjncDPE1bELPFpZkkuZbSdAk has 2 open connections
2023-11-22T12:55:33.626Z backend:libp2p:websockets outbound connection /dns4/ibljzlyaw5cknz2wzducoqofec7z5ogm2de7lfuvu47t2gnei2t7bjqd.onion/tcp/80/ws/p2p/QmNcQbPejv6MDBt9VVt7A9uSU1a2yiYnhHsnVo2hhfqpFL upgraded
2023-11-22T12:55:34.201Z backend:Libp2pService QmYFmZvEikCwJpCjZ6FAiXnjncDPE1bELPFpZkkuZbSdAk ping to QmNcQbPejv6MDBt9VVt7A9uSU1a2yiYnhHsnVo2hhfqpFL. Latency: 2839
2023-11-22T12:55:35.192Z backend:Libp2pService QmYFmZvEikCwJpCjZ6FAiXnjncDPE1bELPFpZkkuZbSdAk disconnected from QmNcQbPejv6MDBt9VVt7A9uSU1a2yiYnhHsnVo2hhfqpFL
2023-11-22T12:55:35.192Z backend:Libp2pService QmYFmZvEikCwJpCjZ6FAiXnjncDPE1bELPFpZkkuZbSdAk has 0 open connections
2023-11-22T12:55:35.193Z backend:Libp2pService QmYFmZvEikCwJpCjZ6FAiXnjncDPE1bELPFpZkkuZbSdAk is connected to 0 peers
node:internal/event_target:1006
  process.nextTick(() => { throw err; });
                           ^
Error: Received wrong ping ack
    at PingService.ping (C:\Users\WinFish\AppData\Local\Programs\@quietdesktop\resources\app.asar\node_modules\backend-bundle\bundle.cjs:453888:32)
    at async Libp2pNode.ping (C:\Users\WinFish\AppData\Local\Programs\@quietdesktop\resources\app.asar\node_modules\backend-bundle\bundle.cjs:459182:16)
    at async Libp2pNode.<anonymous> (C:\Users\WinFish\AppData\Local\Programs\@quietdesktop\resources\app.asar\node_modules\backend-bundle\bundle.cjs:325233:29) {
  code: 'ERR_WRONG_PING_ACK'
```

**Comment by holmesworcester** [comment]
@EmiM I'm very curious if you are able to reproduce this!

**Comment by kingalg** [comment]
@holmesworcester we've already discussed this with Emi before I've created this issue. She knows how to fix it and is almost done with it. (that's why I've assigned it straight to her)

**Comment by kingalg** [comment]
It appears to be more complicated so I'm adding additional info:
1. Error mentioned in an initial post appears when a new peer is joining (it happened in different configurations - once when the user joined the owner and twice when the user joined another user with closed owner).
2. There is another error message that appears when the user is closing the app that previously had issue from point 1. Below error message appears in logs of the user that is closing his own app. 

`  desktop:main:main Event: app.activate +9m
  desktop:main:main Closing window +4s
  desktop:main:main Saved state, closed window +4ms
  desktop:main:main Event mainWindow.closed +1ms
  desktop:main:main Event: app.window-all-closed +0ms
  desktop:main:main:err Backend process returned error Error [ERR_IPC_CHANNEL_CLOSED]: Channel closed
    at new NodeError (node:internal/errors:393:5)
    at target.send (node:internal/child_process:740:16)
    at closeBackendProcess (/Applications/Quiet.app/Contents/Resources/app.asar/dist/main/main.js:277:24)
    at App.<anonymous> (/Applications/Quiet.app/Contents/Resources/app.asar/dist/main/main.js:461:5)
    at App.emit (node:events:525:35) {
  code: 'ERR_IPC_CHANNEL_CLOSED'
} +0ms
  desktop:main:main Backend killed: false, Quitting. +7m
(node:4951) UnhandledPromiseRejectionWarning: Error: mainWindow is on unexpected type null
    at Timeout._onTimeout (/Applications/Quiet.app/Contents/Resources/app.asar/dist/main/main.js:445:23)
    at listOnTimeout (node:internal/timers:564:17)
    at process.processTimers (node:internal/timers:507:7)
(Use `Quiet --trace-warnings ...` to show where the warning was created)
(node:4951) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
rumbleefish@MacBook-Pro-Dawid MacOS % `

**Comment by EmiM** [comment]
This part of the code is not important, it's just libp2p.ping which I added to get extra information about connection latency.
I thought that it's enough to just try-catch this and carry on but the error is also thrown on the other peer's side. I am not sure how to catch this.

I think I'll just remove this for now.

It's hard to reproduce. I imagine it's either because of tor weird hidden-disconnecting behavior OR just we tried to ping peer that disconnected in the meantime.
https://github.com/libp2p/js-libp2p/blob/main/doc/LIMITS.md#stream-limits We may be not consuming data in time also.
